src/utils/request/onUnhandledRequest.ts

```diff
@@ -1,36 +1,162 @@
+  return (request, handler) => {
+    const { mask, method } = handler.getMetaInfo()
+
+    if (mask instanceof RegExp) {
+      return Infinity
+    }
+
+    const hasSameMethod = isStringEqual(request.method, method)
+    // Always treat a handler with the same method as a more similar one.
+    const methodScoreDelta = hasSameMethod ? 0 : 0.25
+    const requestPublicUrl = getPublicUrlFromRequest(request)
+    const score = getStringMatchScore(requestPublicUrl, mask)
+
+    return score + methodScoreDelta
+  }
+}
+
+function getScoreForGraphQLHandler(
+  parsedQuery: ParsedQueryPayload,
+): ScoreGetterFn {
+  return (_, handler) => {
+    if (typeof parsedQuery.operationName === 'undefined') {
+      return Infinity
+    }
+
+    const { operationType, operationName } = handler.getMetaInfo()
+    const hasSameOperationType = parsedQuery.operationType === operationType
+    // Always treat a handler with the same operation type as a more similar one.
+    const operationTypeScoreDelta = hasSameOperationType ? 0 : 0.25
+    const score = getStringMatchScore(parsedQuery.operationName, operationName)
+
+    return score + operationTypeScoreDelta
+  }
+}
+
+function getSuggestedHandler(
+  request: MockedRequest,
+  handlers: RequestHandlersList,
+  getScore: ScoreGetterFn,
+): RequestHandler | undefined {
+  const suggestedHandler = handlers
+    .reduce<RequestHandlerSuggestionList>((acc, handler) => {
+      const score = getScore(request, handler)
+      return acc.concat([[score, handler]])
+    }, [])
+    .sort(([leftScore], [rightScore]) => {
+      return leftScore - rightScore
+    })
+    .find(([score]) => {
+      return score <= 2
+    })
+
+  return suggestedHandler?.[1]
+}
+
 export function onUnhandledRequest(
   request: MockedRequest,
-  handler: OnUnhandledRequest = 'bypass',
+  handlers: RequestHandlersList,
+  strategy: UnhandledRequestStrategy = 'bypass',
 ): void {
-  if (typeof handler === 'function') {
-    handler(request)
+  if (typeof strategy === 'function') {
+    strategy(request)
     return
   }
 
-  const publicUrl = getPublicUrlFromRequest(request)
+  const parsedGraphQLQuery = parseGraphQLRequest(request)
+  const handlerGroups = groupHandlersByType(handlers)
+  const relevantHandlers = parsedGraphQLQuery
+    ? handlerGroups.graphql
+    : handlerGroups.rest
 
-  const message = `captured a ${request.method} ${
-    request.url
-  } request without a corresponding request handler.
+  let handlerSuggestion = ''
+  const suggestedHandler = getSuggestedHandler(
+    request,
+    relevantHandlers,
+    parsedGraphQLQuery
+      ? getScoreForGraphQLHandler(parsedGraphQLQuery)
+      : getScoreForRestHandler(),
+  )
 
-  If you wish to intercept this request, consider creating a request handler for it:
+  if (suggestedHandler) {
+    const metaInfo = suggestedHandler.getMetaInfo()
+    handlerSuggestion = `Did you mean to request "${metaInfo.header}" instead?`
+  }
```

Comment:
Nit, what about declaring `handlerSuggestion`  as a const instead of reassigning it?

```suggestion
const handlerSuggestion = suggestedHandler 
  ? `Did you mean to request "${suggestedHandler.getMetaInfo().header}" instead?` 
  : ''
```